x-pack/filebeat/input/azureblobstorage/job/job.go

```diff
@@ -0,0 +1,137 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+// +build !aix
+
+package job
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+
+	cursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/azureblobstorage/state"
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/azureblobstorage/types"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+type Job interface {
+	Do(ctx context.Context, id string) error
+	Name() string
+	Timestamp() *time.Time
+	Source() *types.Source
+}
+
+type AzureInputJob struct {
+	marker    *string
+	client    *azblob.BlobClient
+	blob      *azblob.BlobItemInternal
+	state     *state.State
+	src       *types.Source
+	publisher cursor.Publisher
+}
+
+// NewAzureInputJob, returns an instance of a job , which is a unit of work that can be assigned to a go routine
+func NewAzureInputJob(client *azblob.BlobClient, blob *azblob.BlobItemInternal,
+	marker *string, state *state.State, src *types.Source, publisher cursor.Publisher) Job {
+
+	return &AzureInputJob{
+		marker:    marker,
+		client:    client,
+		blob:      blob,
+		state:     state,
+		src:       src,
+		publisher: publisher,
+	}
+}
+
+func (aij *AzureInputJob) Do(ctx context.Context, id string) error {
+
+	var event beat.Event
+	msg := mapstr.M{}
+	if types.AllowedContentTypes[*aij.blob.Properties.ContentType] {
+		data, err := aij.extractData(ctx)
+		if err != nil {
+			return fmt.Errorf("job with jobId %s encountered an error : %w", id, err)
+		}
+
+		if _, err := msg.Put("message.containerName", aij.src.ContainerName); err != nil {
```

Comment:
Consider initializing the map here and writing all of the data at once. This way the map can be allocated once.
```
event.Fields = mapstr.M{
    "message": mapstr.M{
      "blobName": aij.blob.Name,
      "content_type": ...,
    },
}
```